src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -1060,6 +1050,20 @@ void CouplingSchemeConfiguration::checkIfDataIsExchanged(
                 dataName);
 }
 
+int CouplingSchemeConfiguration::getWaveformUsedOrder(std::string readDataName) const
+{
+  for (const precice::impl::PtrParticipant &participant : _participantConfig->getParticipants()) {
+    for (auto &dataContext : participant->readDataContexts()) {
+      if (dataContext.getDataName() == readDataName) {
+        int usedOrder = dataContext.getInterpolationOrder();
+        PRECICE_ASSERT(usedOrder >= 0); // ensure that usedOrder was set
+        return usedOrder;
+      }
+    }
+  }
```

Comment:
I still would find it easier to pass `participantName` as an argument here and not loop over all participants. Or does this not work?
Could be more robust and would also be easier to read.

Reply:
I think we actually need both, because the same read data can be used by multiple participants with different waveform order. I modified the function in 193a431a0db4f651fc087e6068440b96a3269e39.